keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv.layers import preprocessing as cv_preprocessing
+from keras_cv.utils import preprocessing as preprocessing_utils
+
+
+class RandAugment(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """RandAugment performs the Rand Augment operation on input images.
+
+    This layer can be thought of as an all in one image augmentation layer.  The policy
+    implemented by this layer has been benchmarked extensively and is effective on a
+    wide variety of datasets.
+
+    The policy operates as follows:
+
+    For each `layer`, the policy selects a random operation from a list of operations.
+    It then samples a random number and if that number is less than
+    `probability_to_apply` applies it to the given image.
+
+    References:
+        - [RandAugment](https://arxiv.org/abs/1909.13719)
+
+    Args:
+        value_range: the range of values the incoming images will have.
+            Represented as a two number tuple written [low, high].
+            This is typically either `[0, 1]` or `[0, 255]` depending
+            on how your preprocessing pipeline is setup.
+        num_layers: the number of layers to use in the rand augment policy.
+        magnitude: the magnitude to use for each of the augmentation.
+        magnitude_standard_deviation: the standard deviation to use when drawing values
+            for the perturbations.
+        probability_to_apply:  the probability to apply an augmentation at each layer.
```

Comment:
Assume this is [0, 1]?

Reply:
That's what I would have expected -- a magnitude between 0 and 1 where 0 is "nothing happens" and 1 is "max distortion". However it seems the default is 5 and the example uses 10, which makes it hard for users to figure it what's the acceptable range and what happens different points of the range.

Reply:
Updated to be [0, 1].  Some of the distortions ignore the value so it's not a true no-op.  That is not entirely possible.  We could update our implementation to use the `blend()` method alongside invert but that would change the operation substantially.  Could be an interesting idea though.